base_layer/wallet_ffi/src/error.rs

```diff
@@ -363,3 +363,29 @@ impl From<SchnorrSignatureError> for LibWalletError {
         }
     }
 }
+
+#[derive(Debug, Error, PartialEq)]
+pub enum TransactionError {
+    #[error("The transaction has an incorrect status: `{0}`")]
+    StatusError(String),
+    #[error("The transaction has the wrong number of kernels: `{0}`")]
+    KernelError(String),
+}
+
+/// This implementation maps the internal TransactionError to a set of LibWalletErrors.
+/// The mapping is explicitly managed here.
+impl From<TransactionError> for LibWalletError {
+    fn from(v: TransactionError) -> Self {
+        error!(target: LOG_TARGET, "{}", format!("{:?}", v));
+        match v {
+            TransactionError::StatusError(_) => Self {
+                code: 640,
+                message: format!("{:?}", v),
```

Comment:
Since `TransactionError` derives from `thiserror`, use Display format.

```suggestion
                message: v.to_string()
```

(trait `ToString` is implemented for all `T: Display`)

Reply:
Thanks @sdbondi will do